core/src/main/java/ar/com/wolox/wolmo/core/fragment/WoloxFragmentHandler.java

```diff
@@ -0,0 +1,85 @@
+package ar.com.wolox.wolmo.core.fragment;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import ar.com.wolox.wolmo.core.R;
+import ar.com.wolox.wolmo.core.presenter.BasePresenter;
+import ar.com.wolox.wolmo.core.util.ToastUtils;
+import butterknife.ButterKnife;
+
+/**
+ * This class is used to separate Wolox Fragments logic so that different subclasses of
+ * Fragment can implement MVP without re-writing this.
+ */
+
+class WoloxFragmentHandler<T extends BasePresenter> {
+
+    private final Fragment mFragment;
+    private final IWoloxFragment<T> mWoloxFragment;
+    private boolean mCreated;
+    private T mPresenter;
+    private boolean mMenuVisible;
+    private boolean mVisible;
+
+    WoloxFragmentHandler(IWoloxFragment<T> woloxFragment) {
+        if(!(woloxFragment instanceof Fragment)) {
+            throw new IllegalArgumentException("WoloxFragment should be a Fragment instance");
```

Comment:
I guess we could use `<T extends Fragment>` to catch this issue in compile time, but we would limiting the developer slightly. Another option would be to make `IWoloxFragment` an abstract class instead of an interface an expect that in the constructor signature. What do you think?